### PR TITLE
Future proofing this test to fail when param/ret modifiers are changed in mono

### PR DESCRIPTION
**Pull request by fir3pho3nixx**
As per [this comment](https://github.com/castleproject/Core/issues/277#issuecomment-312941426)

*** Edited ***

@jonorossi - Please feel free to close if you think it is not relevant.

### CODE REVIEW

**Comment by jonorossi** [comment]
Could you explain to me how your change works, I've not used `Assume` for non-theory tests before and so I expected to see the tests show up under "Inconclusive" in the test summary.

> Test Count: 907, Passed: 896, Failed: 0, Warnings: 0, Inconclusive: 0, Skipped: 11

**Comment by jonorossi** [comment]
Oh, I just realised the rest of the unit tests are using `Assume.That`, I completely missed that before merging those changes.

File: src/Castle.Core.Tests/CustomModifiersTestCase.cs [hunk 1]
```diff
@@ -194,6 +193,8 @@ public void ReflectionReturnsCorrectModoptOnReturnTypeForGeneratedType(string ty
 			Assume.That(CustomModifiersTestCase.customModifiers.ContainsKey(typeNameWithoutSuffix));
 
 			var modopts = this.generatedTypes[typeName].GetMethod("Foo").ReturnParameter.GetOptionalCustomModifiers();
+			Assume.That(modopts.Length > 0); // If this fails on mono/linux we have to revisit the commits and issues for IL method parameter modifiers. https://github.com/castleproject/Core/issues/277
```
Comment (jonorossi) [inline src/Castle.Core.Tests/CustomModifiersTestCase.cs]: > IL method parameter modifiers

The last two tests are for custom modifiers on the return type, probably just write "custom modifiers".

**Comment by fir3pho3nixx** [comment]
@jonorossi - all done :)

**Pull request merged**